src/borg/archive.py

```diff
@@ -11,6 +11,7 @@
 from getpass import getuser
 from io import BytesIO
 from itertools import groupby, zip_longest
+from typing import Iterator  # TODO: remove this if maintainers don't want it
```

Comment:
we'll keep the type hints, but please remove all these related "TODOs".

i am generally happy with these, but they made this PR a bit larger than it would have been without them.